## Carta para Dra. Diana Didatolov

A partir dos dados e da bibilioteca de dados importada do Seaborn e a partir do Github dos datasets também do Seaborn[1]foi realizada uma análise sobre os dados e a divisão entre numéricos e categóricos. 

In [13]:
import seaborn as sns
import matplotlib as plt

### Dow Jones

In [14]:
df_d = sns.load_dataset('dowjones')
df_d

Date   Price
0   1914-12-01   55.00
1   1915-01-01   56.55
2   1915-02-01   56.00
3   1915-03-01   58.30
4   1915-04-01   66.45
..         ...     ...
644 1968-08-01  883.72
645 1968-09-01  922.80
646 1968-10-01  955.47
647 1968-11-01  964.12
648 1968-12-01  965.39

[649 rows x 2 columns]

**Data** : É a data de análise do índice de Dow Jones*.


**Price**: Seria o valor do índice naquela data.

*"O **Dow Jones Industrial Average (DJIA)** é composto por ações de 30 das maiores empresas dos EUA. O peso das ações no índice é determinado pelo valor delas, quanto mais alto, mais representativo é o papel. [...] A longevidade faz do Dow Jones uma ferramenta importante para estudar a evolução do mercado de ações dos EUA nos últimos 126 anos, e para comparar o desempenho das bolsas em diferentes momentos históricos. É também um termômetro dos segmentos mais tradicionais da economia americana, pois as empresas que o compõem são grandes, consolidadas e líderes em suas áreas de atuação." [2]

In [15]:
df_d.describe(include = "all")

Date       Price
count                            649  649.000000
mean   1941-11-30 17:42:48.258859776  290.807319
min              1914-12-01 00:00:00   46.850000
25%              1928-06-01 00:00:00  106.900000
50%              1941-12-01 00:00:00  172.270000
75%              1955-06-01 00:00:00  436.730000
max              1968-12-01 00:00:00  985.930000
std                              NaN  256.062906

Podemos perceber que a coleta de dados, quando dividimos em quartis, foi feita a cada 6 meses e em aproximadamente a cada 14 anos. A primeira coleta de dados foi feita em 1914 e a última em 1968. Quando olhamos para a coluna "price", vemos que o menor valor do índice de dowjones foi 46.8, já o valor mais alto foi 985.93, curiosamente os anos máximos e mínimos não estão vinculados com os menores e maiores valores do índice.

---

### Exercise

In [16]:
df_e = sns.load_dataset('exercise')
df_e

Unnamed: 0  id     diet  pulse    time     kind
0            0   1  low fat     85   1 min     rest
1            1   1  low fat     85  15 min     rest
2            2   1  low fat     88  30 min     rest
3            3   2  low fat     90   1 min     rest
4            4   2  low fat     92  15 min     rest
..         ...  ..      ...    ...     ...      ...
85          85  29   no fat    135  15 min  running
86          86  29   no fat    130  30 min  running
87          87  30   no fat     99   1 min  running
88          88  30   no fat    111  15 min  running
89          89  30   no fat    150  30 min  running

[90 rows x 6 columns]

**id:** É o identificador de cada paciente, em que para alguns foi feito mais de uma medição.(numérico)


**diet:** O teor de gordura da dieta, sendo categórico ordinal.(categórico)


**pulse:** batimento cardíaco médio.(categórico)


**time:** tempo em que a medição foi feita.(numérico)


**kind:** Momento em que as medições forem feitas, analisando se o estado era "rest" ou "running".(categórico)

In [17]:
df_e.dtypes

Unnamed: 0       int64
id               int64
diet          category
pulse            int64
time          category
kind          category
dtype: object

In [18]:
df_e.describe(include = "all") 

Unnamed: 0         id    diet       pulse   time  kind
count    90.000000  90.000000      90   90.000000     90    90
unique         NaN        NaN       2         NaN      3     3
top            NaN        NaN  no fat         NaN  1 min  rest
freq           NaN        NaN      45         NaN     30    30
mean     44.500000  15.500000     NaN   99.700000    NaN   NaN
std      26.124701   8.703932     NaN   14.858471    NaN   NaN
min       0.000000   1.000000     NaN   80.000000    NaN   NaN
25%      22.250000   8.000000     NaN   90.250000    NaN   NaN
50%      44.500000  15.500000     NaN   96.000000    NaN   NaN
75%      66.750000  23.000000     NaN  103.000000    NaN   NaN
max      89.000000  30.000000     NaN  150.000000    NaN   NaN

Pode-se perceber que, a partir da análise pela função dtypes, os dados numéricos não admitem os fatores "unique", "top" e "freq", visto que tais valores não são discretos então não possuem muitas repetições. Além disso, a partir da análise desses mesmos valores numéricos, pode-se notar valores de média, quartis e desvio padrão principalmente da pulsação dos paricipantes (visto que os outros valores numéricos somente indicam a ordem e identificação dos participantes, não sendo colunas passíveis de conclusões por tal análise). Com isso, podemos peceber em uma análise rápida que a maioria tem dieta do tipo "no fat", o que não é verdade já que a frequência pe de 45 vezes sendo que temos 90 dados.

---

### Flights

In [19]:
df_f = sns.load_dataset('flights')
df_f

year month  passengers
0    1949   Jan         112
1    1949   Feb         118
2    1949   Mar         132
3    1949   Apr         129
4    1949   May         121
..    ...   ...         ...
139  1960   Aug         606
140  1960   Sep         508
141  1960   Oct         461
142  1960   Nov         390
143  1960   Dec         432

[144 rows x 3 columns]

**year:** Ano que o vôo aconteceu.(numérico)


**month:** Mês em que o vôo aconteceu.(categórico)


**passengers:** quantidade de passagegeiros naquele vôo.(númerico)

In [20]:
df_f.describe(include = "all")

year month  passengers
count    144.000000   144  144.000000
unique          NaN    12         NaN
top             NaN   Jan         NaN
freq            NaN    12         NaN
mean    1954.500000   NaN  280.298611
std        3.464102   NaN  119.966317
min     1949.000000   NaN  104.000000
25%     1951.750000   NaN  180.000000
50%     1954.500000   NaN  265.500000
75%     1957.250000   NaN  360.500000
max     1960.000000   NaN  622.000000

Os vôos aconteceram em sua maioria em janeiro, que foram 12 vôos ao todo, o que gera uma dúvida, já que temos 144 dados e existem 12 meses no mínimo era necessário para determinar uma moda que um mês tenha aparecido pelo menos 13 vezes, ou seja não temos uma moda. O vôo com menor número de passageiros teve 104 pessoas, e um vôo comportou no máximo 622 pessoas. O mean dos year não faz sentido, pois ele dividiu os anos e dos meses. Um quarto dos vôos aconteceram antes de 1951, e oustro 1/4 dos mêses aconteceram deposis de 1957. Então a maioria dos vôos aconteceram entre 1951 e 1957.

---

## Conclusão

Portanto, percebemos que é muito importante analisar as colunas e como elas interferem nos valores e descrições entre si, pois podem propor ordenamento diferente e interpretações mais profundas e pertinentes. É necessário ter cuidado quado for entender as descrições, pois muitas vezes a moda não é realmete o que representa pois a frequência como exemplificado anteriomente pode ser a mesma para os componentes e mesmo assim o algorítimo propor uma moda. Assim não é sobre encontrar resultados mas ter a sencibilidade de perceber a pertinência deles. 

## Referências
* [1] **GITHUB**. seaborn-data. Disponível em: https://github.com/seaborn/seaborn-data. Acesso em: 26 ago. 2024.
* [2] **INFO MONEY**. O que é o índice Dow Jones e qual a sua importância. Disponível em: <https://www.infomoney.com.br/guias/indice-dow-jones/>. Acesso em: 26 ago. 2024.

